kcfinder/core/class/uploader.php

```diff
@@ -759,11 +768,20 @@ protected function callBack($url, $message="") {
                 $js = $this->$method($url, $message);
         }
 
-        if (!isset($js))
+        if ($this->outputFormat == 'json') {
+          header('Content-Type: application/json');
+          $json = $this->callBack_json($url, $message);
+          echo json_encode($json);
+        }
+        else {
+          $js = $this->callBack_ckeditor($url, $message);
+          if (!isset($js)) {
             $js = $this->callBack_default($url, $message);
 
-        header("Content-Type: text/html; charset={$this->charset}");
-        echo "<html><body>$js</body></html>";
+            header("Content-Type: text/html; charset={$this->charset}");
+            echo "<html><body>$js</body></html>";
+          }
+        }
```

Comment:
There's something odd going on here. The deleted line on 762 has no opening bracket, but the line you aded on 778 does. I think that may be causing the odd behavior.